### PR TITLE
Add background info article on protocol features and characteristics

### PR DESCRIPTION
**Pull request by JakaMohorkoDS**
# Description:

New documentation article that focuses on defining core features, highlighting what server/client protocols support which feature. Additionally, it defines a few key protocol characteristic groups for our protocols, and provides brief notes on incompatibilities.

### CODE REVIEW

**Comment by JakaMohorkoDS** [comment]
Only "protocols_features_characteristics.adoc" needs to be reviewed. Rest is just reorganization.

File: docs/Antora/modules/knowledge_base/pages/protocol_features_characteristics.adoc [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+
+* **Configuration**: The server sends information on its tree structure and properties to the client, allowing the client to produce a mirror of the device's component tree. This allows the client-side application to act as if the device was local to the client instance, traversing its tree, and modifying its property values. A configuration connection does not transfer signal data.
+* **Streaming**: The server sends information on what signals it has available to the client, and sends the data sent through the signals if the client is subscribed to said signals. This form of connection is usually combined with a configuration connection, allowing users to see the whole device structure, as well as read the device's signal data. If only a streaming connection is formed, the client will not see the full component tree, but will still be able to see a list of the device's signals directly underneath the device's "signals" folder.
```
Comment (NikolaiShipilov) [inline docs/Antora/modules/knowledge_base/pages/protocol_features_characteristics.adoc:12]: The sentence seems a bit confusing. I would rephrase it if possible, e.g.: `The server sends to the client: the information on what signals it has available; and the data sent through the signals if the client is subscribed to said signals.`

File: docs/Antora/modules/knowledge_base/pages/protocol_features_characteristics.adoc [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+
+NOTE: For more information on components and properties, you can refer to the xref:components.adoc[Components] and xref:property_system.adoc[Property System] articles.
+
+== Protocol overview
+
+Currently, openDAQ(TM) supports 3 different server/client protocols: Native, OPC UA, and LT
+
+=== Native
+
+The main protocol that is used when an openDAQ(TM) instance is running both on the server and client. The protocol aims to support all core SDK features via the client and aims to minimize network traffic while optimizing performance when connecting openDAT(TM) to openDAQ(TM). It allows for both streaming and configuration connections to be formed. The Native protocol makes extensive use of the SDK's serialization mechanism, transferring the JSON strings obtained when serializing components to the client, which in turn deserializes/updates its own mirrored versions of the server components.
```
Comment (NikolaiShipilov) [inline docs/Antora/modules/knowledge_base/pages/protocol_features_characteristics.adoc:22]: * "... on both the server and client."
* typo: openDAT(TM)  -> openDAQ(TM)

**Review by NikolaiShipilov — APPROVED** [review]
Regarding the protocols details, everything appears accurate; 
I've added just a few stylistic nitpicking refinements.
